src/AccessTokenCacheHandler.php

```diff
@@ -52,6 +52,6 @@ public function deleteItemByProvider(OAuth2Provider $provider, array $options):
 
     public function getCacheKey(OAuth2Provider $provider, array $options): string
     {
-        return static::CACHE_KEY_PREFIX . md5(print_r($provider, true) . serialize($options));
+        return static::CACHE_KEY_PREFIX . md5(get_class($provider) . serialize($options));
```

Comment:
Assume we use the same provider class, but for different services, for example _ProviderX_ for _service A_, and _ProviderX_ for _serviceB_. The different scopes will be handled by the `serialize($options)` part, however due to how the `AbstractProvider` class is structured, the credentials are stored in the provider object.

We would need to avoid returning the cached token if we are performing the request with different credentials.

Why was this returning a different key each request? Would `spl_object_hash` solve the issue? If we could create a test case for this would be great

Reply:
`spl_object_hash`might work since it won't check properties. Since php 7+ is required we could use `spl_object_id` which is supposedly faster.
I will test this approach and let you know if it works on my scenario. But then I don't think either `spl_object_id`or `spl_object_hash`will be able to understand the credentials in the provider changed, worth checking anyway.

Reply:
You're right. Normally we use immutable providers, even if they are instantiations of the same class, however you make a good point here that if the provider is used as a mutable object the cache won't work properly, and that's not desirable.

Could you please provide the scenario where this is not generating the key properly?

Reply:
Taking into account that the AbstractProvider is mutable I think use the `spl_object_hash` or `spl_object_id` is not an option.

Mutable object example:
```
>>> $obj = new \stdClass();
=> {#203}
>>> spl_object_id($obj);
=> 203
>>> spl_object_hash($obj);
=> "000000005860fc47000000005b209971"
>>> $obj->attr = 'test';
=> "test"
>>> spl_object_id($obj);
=> 203
>>> spl_object_hash($obj);
=> "000000005860fc47000000005b209971"
```

`print_r` avoid this issue and create a specific cache key depending on the whole object. Could you clarify with a test what is your issue?